app/services/storage.js

```diff
@@ -213,11 +222,22 @@ export class ProfileStorage {
 
   async startSession(scope: ?number,
                      ancestor: ?number,
-                     now: number = microtime.now()): Promise<number> {
+                     now: number = microtime.now(),
+                     reason: number = SessionStartReason.newTab): Promise<number> {
+    const result =
+      await this.db
+                .run('INSERT INTO sessionStarts (scope, ancestor, ts, reason) VALUES (?, ?, ?, ?)',
+                     [scope, ancestor, now, reason]);
+    return result.lastID;
+  }
+
+  async endSession(session: number,
+                   now: number = microtime.now(),
+                   reason: number = SessionEndReason.tabClosed): Promise<number> {
```

Comment:
I agree with you. But the long-term-thinking way of doing this is to make those two things disjoint spaces — i.e., both "chat closed" and "tab closed" would be members of the same enum. Or a session would be a chat session or a tab session, and the value would be (tab)closed in both cases.